src/category_theory/limits/shapes/regular_mono.lean

```diff
@@ -61,6 +62,37 @@ def regular_mono.lift' {W : C} (f : X ⟶ Y) [regular_mono f] (k : W ⟶ Y)
   {l : W ⟶ X // l ≫ f = k} :=
 fork.is_limit.lift' regular_mono.is_limit _ h
 
+/-- If `h` is a regular mono and `g` is a pullback of `h`, then `g` is a regular mono. -/
+def regular_of_is_pullback_snd_of_regular {P Q R S : C} {f : P ⟶ Q} {g : P ⟶ R} {h : Q ⟶ S} {k : R ⟶ S}
+  [hr : regular_mono h] (comm : f ≫ h = g ≫ k) (t : is_limit (pullback_cone.mk _ _ comm)) :
+regular_mono g :=
+{ Z := hr.Z,
+  left := k ≫ hr.left,
+  right := k ≫ hr.right,
+  w := by rw [← reassoc_of comm, ← reassoc_of comm, hr.w],
+  is_limit :=
+  begin
+    apply fork.is_limit.mk' _ _,
+    intro s,
+    have l₁ : (fork.ι s ≫ k) ≫ regular_mono.left = (fork.ι s ≫ k) ≫ regular_mono.right,
+      rw [category.assoc, s.condition, category.assoc],
+    obtain ⟨l, hl⟩ := fork.is_limit.lift' hr.is_limit _ l₁,
+    obtain ⟨p, hp₁, hp₂⟩ := pullback_cone.is_limit.lift' t _ _ hl,
+    refine ⟨p, hp₂, _⟩,
+    intros m w,
+    have z : m ≫ g = p ≫ g := w.trans hp₂.symm,
+    apply t.hom_ext,
+    apply (pullback_cone.mk f g comm).equalizer_ext,
+    { erw [← cancel_mono h, category.assoc, category.assoc, comm, reassoc_of z] },
+    { exact z },
+  end }
+
+/-- If `k` is a regular mono and `f` is a pullback of `k`, then `f` is a regular mono. -/
+def regular_of_is_pullback_fst_of_regular {P Q R S : C} {f : P ⟶ Q} {g : P ⟶ R} {h : Q ⟶ S} {k : R ⟶ S}
```

Comment:
I'm still not happy with the naming and the doc-strings differentiating these two lemmas.

In particular, the two doc-strings are alpha-equivalent at the moment.